flink-connectors/flink-jdbc/src/main/java/org/apache/flink/api/java/io/jdbc/JDBCTableSource.java

```diff
@@ -74,12 +118,56 @@ public static Builder builder() {
 		return new Builder();
 	}
 
+	private JDBCInputFormat getInputFormat() {
+		JDBCInputFormat.JDBCInputFormatBuilder builder = JDBCInputFormat.buildJDBCInputFormat()
+				.setDrivername(options.getDriverName())
+				.setDBUrl(options.getDbURL())
+				.setUsername(options.getUsername())
+				.setPassword(options.getPassword())
+				.setRowTypeInfo(new RowTypeInfo(schema.getFieldTypes(), schema.getFieldNames()));
+
+		String query = options.getDialect().getSelectFromStatement(
+			options.getTableName(), returnType.getFieldNames(), new String[0]);
+		if (scanOptions != null) {
+			long lowerBound = scanOptions.getPartitionLowerBound();
+			long upperBound = scanOptions.getPartitionUpperBound();
+			long numPartitions = scanOptions.getNumPartitions();
+			// partitionSize = Math.ceil(upperBound - lowerBound + 1) / numPartitions;
+			// the following is equivalent
```

Comment:
If they are equal, why not use `Math.ceil` ? And it seems that they are not equal.

Reply:
I change the interfaces of `NumericBetweenParametersProvider`. We do not need `Math.ceil` any more.